### PR TITLE
Impl of `/browse/object/field` endpoint

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
This goes into `/feature/browse_api` long-lived branch.

This implements https://radixdlt.atlassian.net/wiki/spaces/RPNV1/pages/3189211214/RNP-12+-+Browse+API+sub-API+within+Core+API#4.-Object-Field-Lookup

### CODE REVIEW

**Comment by jakrawcz-rdx** [comment]
Today @iamyulong made me realize that this PR has no support for fields of non-instantiated accounts/identities.

@dhedey - I assume we want such support. If so, it can be added in some next PR (maybe it affects other endpoints too). Or should we rather keep returning "object not found" for /object/field?

**Comment by dhedey** [comment]
@jakrawcz-rdx yes we need to support virtual accounts and identities as if they existed. Arguably the virtualisation could/should be in the system reader. Happy if this is in a later PR

**Review by malexandru-rdx — APPROVED** [review]
LGTM!

File: core-rust/core-api-server/src/core_api/handlers/browse_object_field.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            Ok(FieldCoordinate::Name(field_name))
+        }
+    } else if let Some(field_index) = field_index {
+        Ok(FieldCoordinate::Index(extract_api_u8_as_i32(field_index)?))
+    } else {
+        Err(ExtractionError::InvalidFieldAlternativesUsage)
+    }
+}
+
+enum FieldCoordinate {
```
[the remark this answers is not in the document]
Comment (jakrawcz-rdx) [reply core-rust/core-api-server/src/core_api/handlers/browse_object_field.rs:63]: I thought about a couple of names, and all ideas are suboptimal (e.g. the "field address" clashes with our other notion of address [i.e. package address, component address, resource address] which operates on the ReNode layer and has bech32m encoding, and thus is clearly something else).

LUCKILY, the `FieldCoordinate` totally goes away in [one of the next PRs](https://github.com/radixdlt/babylon-node/pull/741/files#diff-ceeb11b11fda04e2678b4523264d792237edbeef51f1d0ae3128f33423b57c02L63) :D

image

... so we don't have to worry about its name too much 🤷 

For those curious:
The {index + name} concept was expanded not only to "fields" but also to "collections". And the common notion for this (that I came up with) is "rich index" (rationale: we have a numeric index, which may be enriched by a human-readable name). So in the field's/collection's metadata output we have `RichIndex{num + Optional<str>}`, and on the input we have `RichIndexInput{num | str}`. Ofc we can discuss this naming too.

**Pull request merged**